Compile errors on GCC 4 from PR 736

### Describe the problem you're observing
UnifyFS is now failing to build on the default compiler on LC systems (GCC v4.9.3) since PR #736 was merged.

The issues are stemming from `unifyfs-ls.cpp`.

### Describe how to reproduce the problem

Build UnifyFS using GCC v4.9.3

First failed in GitLab CI, reproduced manually as well.

### Include any warning or errors or releveant debugging data

```
../../../../util/unifyfs-api-client/src/unifyfs-ls.cpp: In function ‘int main(int, char**)’:
../../../../util/unifyfs-api-client/src/unifyfs-ls.cpp:95:10: error: ‘it’ does not name a type
     auto it = file_set.cbegin();
          ^
../../../../util/unifyfs-api-client/src/unifyfs-ls.cpp:96:12: error: ‘it’ was not declared in this scope
     while (it != file_set.cend()) {
            ^
../../../../util/unifyfs-api-client/src/unifyfs-ls.cpp:96:27: error: ‘class std::set<unifyfs_server_file_meta, CompareByFilename>’ has no member named ‘cend’
     while (it != file_set.cend()) {
                           ^
../../../../util/unifyfs-api-client/src/unifyfs-ls.cpp:109:5: error: ‘it’ was not declared in this scope
     it = file_set.cbegin();
     ^
../../../../util/unifyfs-api-client/src/unifyfs-ls.cpp:109:19: error: ‘class std::set<unifyfs_server_file_meta, CompareByFilename>’ has no member named ‘cbegin’
     it = file_set.cbegin();
                   ^
../../../../util/unifyfs-api-client/src/unifyfs-ls.cpp:111:26: error: ‘free’ was not declared in this scope
         free(it->filename);
                          ^
../../../../util/unifyfs-api-client/src/unifyfs-ls.cpp: In function ‘void parseOpts(int, char**, CommandOptions&)’:
../../../../util/unifyfs-api-client/src/unifyfs-ls.cpp:194:28: error: ‘exit’ was not declared in this scope
                     exit( 0);
                            ^
make[4]: *** [Makefile:639: unifyfs_ls-unifyfs-ls.o] Error 1
```

--- a/util/unifyfs-api-client/src/Makefile.am
+++ b/util/unifyfs-api-client/src/Makefile.am
@@ -16,6 +16,10 @@ api_client_cppflags = \
    -I$(top_srcdir)/client/src \
    -I$(top_srcdir)/common/src
 
+api_client_cxxflags = \
+   $(AM_CXXFLAGS) \
+   -std=c++11
+
 api_client_ldadd   = $(top_builddir)/client/src/libunifyfs_api.la
 api_client_ldflags = $(AM_LDFLAGS) -static
 
@@ -27,6 +31,7 @@ unifyfs_laminate_LDFLAGS  = $(api_client_ldflags)
 unifyfs_laminate_SOURCES  = unifyfs-laminate.c
 
 unifyfs_ls_CPPFLAGS = $(api_client_cppflags)
+unifyfs_ls_CXXFLAGS = $(api_client_cxxflags)
 unifyfs_ls_LDADD    = $(api_client_ldadd)
 unifyfs_ls_LDFLAGS  = $(api_client_ldflags)
 unifyfs_ls_SOURCES  = unifyfs-ls.cpp

